reps/2023-03-20-air-storage-path.md

```diff
@@ -0,0 +1,151 @@
+## Summary
+
+We'd like to change how users of Ray AIR (specifically: Train + Tune) configure permanent storage locations.
+
+Current way:
+- Users can specify `RunConfig.local_dir` or `tune.run(local_dir)` for storage on the local node
+- Users can specify `SyncConfig.upload_dir` for storage on remote nodes
+
+Future way:
+- We have a unified setting `RunConfig.storage_path` that can be set either to a local dir or a remote path.
+- Setting the storage path to a cloud or NFS URI (e.g., `s3://`, or `file://` that points to a NFS mount). In these cases, data will be first written to a local cache dir on the worker, and then synced to a subdirectory in the storage path designated by `<experiment_name>/<trial_name>/`.
+- Setting the storage path to a purely local URI (e.g., `/home/foo/ray_results`). In this mode, there is no persistence once nodes die, nor is there any syncing. The user will have to figure out which node their result is stored on in order to retrieve their result manually. We would only recommend this mode for single node cluster operation generally.
```

Comment:
Should we still support the "pure local URI" to work with multiple nodes? Same question as Eric below, not sure if this is going to be higher impact than we would like...

Reply:
Actually I am reading through the PRD and I'm not seeing that we are actually intending to disable syncing to the head node. Are you seeing anything that says this differently?

Reply:
This was based on an offline discussion. But I generally agree, the syncing/not syncing part is orthogonal. I'll remove this from the PRD and we keep the current node-to-node syncing behavior.